### ISSUE TITLE
Unable to install Couchbase extention on ubuntu 20.04 & php 8.0

### ISSUE THREAD

**Issue by StuMason** [shivammathur/setup-php#593]
**Describe the bug**
The following stanza was working previously for us: 

```yaml
     - name: Setup PHP with PECL extension
        uses: shivammathur/setup-php@v2
        with:
          php-version: "8.0"
          extensions: couchbase
        env:
          fail-fast: true
```

However this is now failing, it's taking around 15 minutes to complete, and gives the following error: 

```
==> Setup Extensions
PHP Warning:  PHP Startup: Invalid library (maybe not a PHP library) 'libcouchbase_php_core.so' in Unknown on line 0
✗ couchbase Could not install couchbase on PHP 8.0.18
```

**Version**
- [ ] I have checked releases, and the bug exists in the latest patch version of `v1` or `v2`.
- [x] `v2`
- [ ] `v1`

**Runners**

- [x] GitHub Hosted
- [ ] Self Hosted

**Operating systems**

Ubuntu 20.04

**PHP versions**

8.0, 8.1

**To Reproduce**

```
     - name: Setup PHP with PECL extension
        uses: shivammathur/setup-php@v2
        with:
          php-version: "8.1"
          extensions: couchbase
        env:
          fail-fast: true
```

**Expected behavior**

Installs php with couchbase extension enabled

**Screenshots/Logs**

```
/usr/bin/bash /home/runner/work/_actions/shivammathur/setup-php/v2/src/scripts/run.sh

==> Setup PHP
✓ PHP Switched to PHP [8](https://github.com/pfizer/divine-waterfall/runs/6453536857?check_suite_focus=true#step:5:8).0.18

==> Setup Extensions
PHP Warning:  PHP Startup: Invalid library (maybe not a PHP library) 'libcouchbase_php_core.so' in Unknown on line 0
✗ couchbase Could not install couchbase on PHP 8.0.18

==> Setup Tools
PHP Warning:  PHP Startup: Invalid library (maybe not a PHP library) 'libcouchbase_php_core.so' in Unknown on line 0
✓ composer Added composer 2.3.5

==> #StandWithUkraine
✓ read-more https://setup-php.com/support-ukraine
```

![image](https://user-images.githubusercontent.com/21249318/168618316-daa4b379-4f13-4ed9-821f-7d23f7440444.png)

![image](https://user-images.githubusercontent.com/21249318/168618451-006139e6-b9c5-4fdc-b829-f8a744edcb0b.png)

**Additional context**

Last successful run was on the 29th March 2022

**Are you willing to submit a PR?**

With help, yes.

**Comment by StuMason** [comment]
fwiw, it's just worked on Ubuntu 22.04 after 10 minutes:

![image](https://user-images.githubusercontent.com/21249318/168619537-bac4d10c-326d-4881-bb71-9e271992da53.png)

**Comment by polesen** [comment]
I am getting the same problem with PHP7.4 on ubuntu-20.04. I *have* to stay on the v2 of libcouchbase/couchbase extension because v3 and v4 are not API compatible.

I suspect, that upgrading to ubuntu-22.04 will give me v4 right? And if so, is there a way to stay on couchbase extension v2? I tried changing my extensions option to include "couchbase-2" but that did not help.

**Comment by shivammathur** [comment]
In the next release, instead of upgrading to v4 for now, I will wait till v4 becomes a bit stable as it fails to compile on some configurations. So it will be v2 till PHP 7.1 and v3 for 7.2 and above by default.

But, you will be able to specify the exact version of couchbase you want after the next release as per the example below. I plan to do that on Monday next week.

```yml
- name: Setup PHP
  uses: shivammathur@setup-php@v2
  with:
    php-version: '7.4'
    extensions: couchbase-2.6.2
```

**Comment by shivammathur** [comment]
Released 2.19.0 with the fix. 

I managed to add support for couchbase 4. So that will be default on PHP 7.4 and above. But you can specify the exact semver version in extensions input if you want an older version.

https://github.com/shivammathur/setup-php/releases/tag/2.19.0
